packages/core/src/browser/preferences/preference-proxy.ts

```diff
@@ -86,15 +86,7 @@ export function createPreferenceProxy<T>(preferences: PreferenceService, schema:
         const preferenceName = OverridePreferenceName.is(arg) ?
             preferences.overridePreferenceName(arg) :
             <string>arg;
-        const value = preferences.get(preferenceName, defaultValue, resourceUri || opts.resourceUri);
-        if (preferences.validate(preferenceName, value)) {
-            return value;
-        }
-        if (defaultValue !== undefined) {
-            return defaultValue;
-        }
-        const values = preferences.inspect(preferenceName, resourceUri);
-        return values && values.defaultValue;
+        return preferences.get(preferenceName, defaultValue, resourceUri || opts.resourceUri);
```

Comment:
So we are fine with runtime errors for statically typed proxies? None of clients is checking for invalid values.

Reply:
Yes, it's strange to do it partly and VS code is not doing it at all.